.travis.yml

```diff
@@ -130,7 +130,7 @@ jobs:
       # scenario of docker devfile url testing needs only Kube config file. So the test has been
       # added here just to make sure docker devfile url command test gets a proper kube config file.
       # without creating a separate OpenShift cluster.
-      name: "devfile catalog, create, push, delete and docker devfile url command integration tests"
+      name: "devfile catalog, create, push, delete, registry and docker devfile url command integration tests"
```

Comment:
I think we should split off the devfile integration tests in Travis like so:
- Integration tests that require Kube (odo create, push, delete, url for Kube)
- Integration tests that require Docker (create, push, delete, url for Docker)
- Integration tests that don't require a specific runtime (catalog and registry)

I'm not sure if we should lumping in six separate integration tests scenarios into one travis job, especially since some (catalog and registry) aren't specific to this Docker job

Reply:
I think @amitkrout  already did some categorizing integration tests work:
- All k8s devfile integration tests (with and without k8s cluster) are in this job, seems like he uses `odo login/logout` to differentiate them
- All docker devfile integration tests are here: https://github.com/openshift/odo/blob/master/.travis.yml#L164 (`docker devfile url` is a special case in k8s devfile integration tests, reason is in his comments above)

Reply:
Also I find another job for running k8s cluster only: https://github.com/openshift/odo/blob/master/.travis.yml#L172, the one we added before is running on OpenShift cluster. I think we should open a separate issue only for categorizing tests properly, and leave comments there.

Reply:
Yes, we have them separated or docker and k8s, but I don't think the `registry` test suite that you're adding belongs in the docker devfile test job for travis

Reply:
You are right, I can create a new job to separate test suites that only needs odo runtime, for now the test suites would be `odo catalog`, `odo registry` and `odo create`.

In the mean time, more comments regarding categorizing tests can be discussed under this issue: https://github.com/openshift/odo/issues/3111

Reply:
But I am gonna use separate PR to cover the tests separation as we need to change both travis config file and test structure to remove stuff (eg. namespace) that force us to use cluster inside each test case. So for the time being I will leave `registry` test suite in k8s job and do all the test separation work together in another PR.